### PR TITLE
Css transforms

### PR DESCRIPTION
**Pull request by nerdgilbert**
This commit primarily ports the jQuery animations in the new-ui plugin to use CSS transforms in order to provide the user with smoother and more efficient animations of gauges and indicators. In addition, it completely replaces the artificial horizon navigation plugin with an attitude indicator plugin that significantly increases the CPU idle percentage. Finally, it also contains changes for indicating if the ROV is using the compass or gyro to navigate and adjusts the UI accordingly.

### CODE REVIEW

File: src/plugins/new-ui/public/webcomponents/compass.html [hunk 1]
```diff
@@ -48,143 +84,265 @@
...
+        listeners: {
+          'mode.tap': 'toggleMode',
+          'heading.tap': 'toggleHeadingLock'
+        },
+        toggleMode: function(e) {
+
+          //This is a toggle
+          if(isCompass)
+          {
+            this.$.mode.textContent = "Gyro";
```
Comment (BrianAdams) [inline src/plugins/new-ui/public/webcomponents/compass.html]: You should simply set the property and let the data binding in the template take care of the rest.  Simply `this.set('mode','Gyro');`

Comment (BrianAdams) [inline src/plugins/new-ui/public/webcomponents/compass.html]: However, on reading further you are using a state isCompass to figure out what is happening. So you should simply create a helper method _compassMode() for instance, and use that in the template {{__(_compassMode())}} where the helper checks isCompass and return the right text.

File: src/plugins/new-ui/public/webcomponents/compass.html [hunk 2]
```diff
@@ -48,143 +84,265 @@
...
+            $(this.$.gyroIndicators).css({display:'block'});
+
+            isCompass = false;
+          }
+          else
+          {
+            this.$.mode.textContent = "Compass";
+
+            //Display svg elements on the rose that show heading
+            $(this.$.compassIndicators).css({display:'block'});
```
Comment (BrianAdams) [inline src/plugins/new-ui/public/webcomponents/compass.html]: A better way to do this is to use the Hidden attribute provided by Polymer. You simply create a helper function and bind that to the hidden attribute of the compassIndicator.

File: src/plugins/translations.en-US/new-ui.json [hunk 3]
```diff
@@ -22,5 +22,6 @@
   "This page allows interaction with past recoded dives.": "This page allows interaction with past recoded dives.",
   "ROV Offline": "ROV Offline",
   "SwitchesNew": "SwitchesNew",
-  "This feature is in BETA. Please report problems at https://github.com/openrov/openrov-software/issues": "This feature is in BETA. Please report problems at https://github.com/openrov/openrov-software/issues"
+  "This feature is in BETA. Please report problems at https://github.com/openrov/openrov-software/issues": "This feature is in BETA. Please report problems at https://github.com/openrov/openrov-software/issues",
+  "This is a test": "This is a test"
```
Comment (BrianAdams) [inline src/plugins/translations.en-US/new-ui.json]: Kill the "This is a Test" text from the translation file

File: .vscode/launch.json [hunk 4]
```diff
@@ -47,4 +47,4 @@
         
 		}
 	]
-}
\ No newline at end of file
+}
```
Comment (BrianAdams) [inline .vscode/launch.json]: Be sure to avoid checking in changed to shared file unless they are intended

File: .vscode/settings.json [hunk 5]
```diff
@@ -1,3 +0,0 @@
-// Place your settings in this file to overwrite default and user settings.
```
Comment (BrianAdams) [inline .vscode/settings.json]: Same story here. Go ahead and undo the change to this file

File: src/plugins/navigation-data/public/webcomponents/attitude-indicator.html [hunk 6]
```diff
@@ -0,0 +1,632 @@
...
+       var isDemo = false;
+       var demoCounter = 0;
+
+      (function () {    
+      Polymer({
+        is: 'attitude-indicator',
+        properties: {
+          pitchValue: {
+            type: Number,
+            value: 0
```
Comment (BrianAdams) [inline src/plugins/navigation-data/public/webcomponents/attitude-indicator.html]: Be sure to set the default values to a function that returns the default instead of an actual value to make it individual for each instance of the control

File: src/plugins/new-ui/cockpit.ejs [hunk 7]
```diff
@@ -56,7 +57,13 @@
                     <div id='twitchembed' style='position:absolute; top:0px; left:0px; z-index:33;'></div>
                     <orov-twitch-stream-status style='position:absolute; top:0px; left:0px; z-index:33;' event-emitter="{{cockpitEventEmitter}}"></orov-twitch-stream-status>
 
-                    <orov-horizon id="horizon" style='position:absolute; top:0px; left:0px; z-index:33; width:100%; height:100%' event-emitter="{{cockpitEventEmitter}}"></orov-horizon>
+                    <!--<orov-horizon id="horizon" style='position:absolute; top:0px; left:0px; z-index:33; width:100%; height:100%' event-emitter="{{cockpitEventEmitter}}"></orov-horizon>
```
Comment (BrianAdams) [inline src/plugins/new-ui/cockpit.ejs]: Be sure to remove the commented out code

File: src/plugins/new-ui/public/webcomponents/compass.html [hunk 8]
```diff
@@ -48,143 +84,265 @@
...
           line-height="125%" word-spacing="0" y="530" x="460.67117" font-weight="bold">{{heading}}</text>
-
+          
           <text id="target" fill$="{{ options.bezel }}" style="letter-spacing:0;text-anchor:middle;word-spacing:0;text-align:center;" font-family="Helvetica Neue,Helvetica,Arial,sans-serif" font-size="150px" letter-spacing="0" line-height="100%" word-spacing="0" y="680" x="460.67117" font-weight="bold"
           title="Heading hold target">{{target}}</text>
 
+          <text id="mode" fill$="{{ options.modeFill }}" stroke$="{{ options.bezel }}" stroke-width$="{{ options.headingStrokeWidth }}" style="letter-spacing:0;text-anchor:middle;word-spacing:0;text-align:center;" font-family="Helvetica Neue,Helvetica,Arial,sans-serif" font-size="200px" letter-spacing="0"
+          line-height="125%" word-spacing="0" y="1206" x="460.67117" font-weight="bold" tabindex="0">{{__(mode)}}</text>
```
Comment (BrianAdams) [inline src/plugins/new-ui/public/webcomponents/compass.html]: Per other comments, this should be a computed value via a helper function that looks at the isCompass boolean to determine the label text.

File: src/plugins/new-ui/public/webcomponents/compass.html [hunk 9]
```diff
@@ -48,143 +84,265 @@
...
-      is: 'orov-newui-compass',
-      properties: {
-        bearing: {
-          observer: 'setBearing',
-          notify: true
+      var lastData = null;
+      var newData = null;
+
+      //The operating mode of the display
+      var isCompass = true;
```
Comment (BrianAdams) [inline src/plugins/new-ui/public/webcomponents/compass.html]: I'd go ahead and make this a first class property. In that sense, the user could default the mode by setting this attribute in the HTML.

File: src/plugins/new-ui/public/webcomponents/compass.html [hunk 10]
```diff
@@ -48,143 +84,265 @@
...
-          type: String,
-          value: '000',
-          observer: 'setHeading',
-          notify: true
+        toggleHeadingLock: function(e) {
+          
+          //This is a toggle
+          if(this.$.headingLock)
+          {
+            console.log("Heading lock disabled")
```
Comment (BrianAdams) [inline src/plugins/new-ui/public/webcomponents/compass.html]: Kill the log statements

File: src/plugins/navigation-data/public/webcomponents/attitude-indicator.html [hunk 11]
```diff
@@ -0,0 +1,632 @@
...
+            }
+            else if(this.pitch < -90)
+            {
+              yPos = -90;
+            }
+            else
+            {
+              yPos = map(this.pitch, -90, 90, -287, 112);
+            }
+            this.$.crosshair.style.transform = "translate3d(-117px," + yPos + "px, 0px) rotate(" + this.roll + "deg)";
```
Comment (BrianAdams) [inline src/plugins/navigation-data/public/webcomponents/attitude-indicator.html]: This code is a little hard to ready. If you simply moved the code to a helper function mapPitchToCrossHairYPos() that would help a lot.

**Comment by nerdgilbert** [comment]
Made recommended changes except for using Polymer hidden element for hiding the compass indicators because it was not working to svg elements (e.g. <g></g> tags). Open for explanation on the most idiomatic way to achieve the same functionality.

**Comment by nerdgilbert** [comment]
Ready for review
